[AssetMapper] Add info about troubleshooting assets loading with Panther.

WedgeSama opened the pull request:
Add more info on Panther tests due to PHP built in server.

When a requested URI is not a `.php` file nor a directory, the built in server will try to directly server the corresponding "file". But if the file does not exist AND must be handle by Symfony app, the built in server just return a 404.

- Can happen with any requested URI that look like a non `.php` file
- Case can happen with AssetMapper.


see
https://github.com/symfony/panther/issues/630
https://symfonycasts.com/screencast/last-stack/testing#debugging-by-opening-the-browser
https://www.php.net/manual/en/features.commandline.webserver.php

MrYamous commented on testing/end_to_end.rst:
```diff
@@ -799,6 +799,56 @@ variable to ``false`` in your style file:
+~~~~~~~~~~~~~~~~~~
+
+You may face cases where your assets are not loaded while running your tests.
+Because Panther use the `PHP built in server`_ to serve your app, if your assets files
```
```suggestion
Because Panther use the `PHP built-in server`_ to serve your app, if your assets files
```

MrYamous commented on testing/end_to_end.rst:
```diff
@@ -799,6 +799,56 @@ variable to ``false`` in your style file:
+You may face cases where your assets are not loaded while running your tests.
+Because Panther use the `PHP built in server`_ to serve your app, if your assets files
+(or any requested URI that not a ``.php`` file) does not exist in your public directory
+(e.g. rendered by your Symfony app), the built in server will return a 404 not found.
```
```suggestion
(e.g. rendered by your Symfony app), the built-in server will return a 404 not found.
```

MrYamous commented on testing/end_to_end.rst:
```diff
@@ -799,6 +799,56 @@ variable to ``false`` in your style file:
+
+    When using :doc:`AssetMapper component </frontend/asset_mapper>`, you can also compile your assets before running
+    your tests. It will make your tests faster because Symfony will not have to handle them, the built in server
+    will served them directly.
```
```suggestion
    will serve them directly.
```

MrYamous commented on testing/end_to_end.rst:
```diff
@@ -799,6 +799,56 @@ variable to ``false`` in your style file:
+.. note::
+
+    When using :doc:`AssetMapper component </frontend/asset_mapper>`, you can also compile your assets before running
+    your tests. It will make your tests faster because Symfony will not have to handle them, the built in server
```
```suggestion
    your tests. It will make your tests faster because Symfony will not have to handle them, the built-in server
```

MrYamous commented on testing/end_to_end.rst:
```diff
@@ -799,6 +799,56 @@ variable to ``false`` in your style file:
 
     $enable-smooth-scroll: false;
 
+Assets not loading
```
Maybe we can use a more specific title as this problem only concern PHP build-in server ?

WedgeSama replied:
Yes you are right, its only when using built-in server.

WedgeSama committed:
Add info about troubleshooting assets loading with Panther.
- Can happen with any requested uri that look like a non `.php` file
- Case can happen with AssetMapper.

javiereguiluz commented:
Benjamin, thanks a lot for this contribution (thanks also to Matthieu for the review too).

While merging we did some changes: https://github.com/symfony/symfony-docs/commit/78e0a190a821e952b3b1ff7b9352060e6720e53d

It looks like a lot of changes 😢 but don't worry. They are just minor cosmetic changes and it rearranges some contents. The core of your contribution was kept. Thanks!
